AmazingMeSample/app/src/main/java/amazingme/activities/app/RegisterActivity.java

```diff
@@ -0,0 +1,60 @@
+package amazingme.activities.app;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+
+import com.amazingme.activities.R;
+
+import amazingme.app.EnumeratedActivity;
+import amazingme.database.FirebaseHelper;
+import amazingme.model.AmazingMeAppCompatActivity;
+
+/**
+ * Created by yudawinata on 8/31/17.
+ */
+
+public class RegisterActivity extends AmazingMeAppCompatActivity {
+    private EditText firstNameEditText, lastNameEditText, emailEditText, passwordEditText;
+    private Button registerBtn, backBtn;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_register);
+
+        firstNameEditText = (EditText) findViewById(R.id.firstNameEditText);
+        lastNameEditText = (EditText) findViewById(R.id.lastNameEditText);
+        emailEditText = (EditText) findViewById(R.id.emailEditText);
+        passwordEditText = (EditText) findViewById(R.id.passwordEditText);
+        registerBtn = (Button) findViewById(R.id.registerBtn);
+        backBtn = (Button) findViewById(R.id.backBtn);
+
+
+        registerBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String firstName = firstNameEditText.getText().toString();
+                String lastName = lastNameEditText.getText().toString();
+                String email = emailEditText.getText().toString();
+                String password = passwordEditText.getText().toString();
+                FirebaseHelper.createNewUser(firstName, lastName, email, password, getApplicationContext());
```

Comment:
i might abstract this a bit more in the next few days. but in the mean time i think it'll be good. good job in filling the firebasehelper class out